cli/command/trust/info.go

```diff
@@ -2,18 +2,62 @@ package trust
 
 import (
 	"context"
-	"encoding/base64"
+	"encoding/hex"
 	"fmt"
+	"sort"
+	"strings"
 
 	"github.com/docker/cli/cli"
 	"github.com/docker/cli/cli/command"
 	"github.com/docker/cli/cli/trust"
 	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/registry"
 	"github.com/docker/notary"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
 	"github.com/spf13/cobra"
 )
 
+const releasedRoleName = "owner"
+
+type trustTagKey struct {
+	TagName string
+	HashHex string
+}
+
+type trustTagRow struct {
+	trustTagKey
+	Signers []string
+}
+
+type trustTagRowList []trustTagRow
+
+func (tagComparator trustTagRowList) Len() int {
+	return len(tagComparator)
+}
+
+func (tagComparator trustTagRowList) Less(i, j int) bool {
+	return tagComparator[i].TagName < tagComparator[j].TagName
+}
+
+func (tagComparator trustTagRowList) Swap(i, j int) {
+	tagComparator[i], tagComparator[j] = tagComparator[j], tagComparator[i]
+}
+
+// check if a role name is "released": either targets/releases or targets TUF roles
+func isReleasedTarget(role data.RoleName) bool {
+	return role == data.CanonicalTargetsRole || role == trust.ReleasesRole
+}
+
+// convert TUF role name to a human-understandable signer name
+func notaryRoleToSigner(tufRole data.RoleName) string {
+	//  don't show a signer for "targets" or "targets/releases"
+	if isReleasedTarget(data.RoleName(tufRole.String())) {
+		return releasedRoleName
+	}
+	return strings.TrimPrefix(tufRole.String(), "targets/")
```

Comment:
Should we do any validation for `tufRole` here? I see that `TrimPrefix` will return the string unchanged if it doesn't start with `targets/`, I wonder if we should return an error or some sort of message in that case, or just leave it be, silently.

(I don't have a strong request for change here, just thinking out loud)